src/components/Audiences/processDestinations.js

```diff
@@ -18,41 +18,38 @@ import {
 } from "../../utils";
 
 export default ({ destinations, config, logger }) => {
-  const urlDestinations = destinations
-    .filter(dest => dest.type === "url")
-    .map(dest =>
-      assign(
-        {
-          id: dest.id
-        },
-        dest.spec
-      )
-    );
+  if (config.destinationsEnabled === undefined || config.destinationsEnabled) {
+    const urlDestinations = destinations
+      .filter(dest => dest.type === "url")
+      .map(dest =>
+        assign(
+          {
+            id: dest.id
+          },
+          dest.spec
+        )
+      );
 
-  if (
-    urlDestinations.length &&
-    (config.destinationsEnabled === undefined || config.destinationsEnabled)
-  ) {
-    fireDestinations({
-      logger,
-      destinations: urlDestinations
-    }).then(result => {
-      console.log(result);
-    });
-  }
-
-  const cookieDestinations = destinations
-    .filter(dest => dest.type === "cookie")
-    .map(dest => dest.spec);
-
-  cookieDestinations.forEach(dest => {
-    if (isNonEmptyString(dest.name)) {
-      cookie.set(dest.name, dest.value || "", {
-        domain: dest.domain || "",
-        expires: dest.ttl ? dest.ttl : 6 * 30 // default of 6 months
+    if (urlDestinations.length > 0) {
+      fireDestinations({
+        logger,
+        destinations: urlDestinations
       });
-    } else {
-      logger.error("Cookie destination had an invalid or no name.");
     }
-  });
+
+    const cookieDestinations = destinations
+      .filter(dest => dest.type === "cookie")
+      .map(dest => dest.spec);
+
+    cookieDestinations.forEach(dest => {
+      if (isNonEmptyString(dest.name)) {
+        cookie.set(dest.name, dest.value || "", {
+          domain: dest.domain || "",
+          expires: dest.ttl ? dest.ttl : 6 * 30 // default of 6 months
+        });
+      } else {
+        logger.error("Cookie destination had an invalid or no name.");
```

Comment:
How/when would the destination not have a name? I think this is one of those things that should be validated when the user is saving destinations and not when we're using them. We should be able to assume that if we get destinations from the gateway, that they are valid and well-structured.